cmd/juju/application/deployer/deployer.go

```diff
@@ -287,11 +288,6 @@ func (d *factory) determineBaseForLocalCharm(charmOrBundle string, getter ModelC
 		imageStream  string
 		selectedBase corebase.Base
 	)
-	ch, err := d.charmReader.ReadCharm(charmOrBundle)
-	if err != nil {
-		return corebase.Base{}, "", nil
-	}
```

Comment:
Turns out when deploying a local charm we used to read the charm from disk twice

Reply:
This was the only invocation of `charmReader` in the codebase. Now that it's been dropped, I have replace the moethod on it's method with `NewCharmAtPath`, and used `charmReader` where before we just concretely called `corecharm.NewCharmAtPath`